src/mpi/comm/commutil.c

```diff
@@ -319,7 +319,11 @@ int MPII_Setup_intercomm_localcomm(MPIR_Comm * intercomm_ptr)
     intercomm_ptr->local_comm = localcomm_ptr;
 
     /* sets up the SMP-aware sub-communicators and tables */
+    /* This routine maybe used inside MPI_Comm_idup, so we can't synchronize
+     * seq using blocking collectives, thus a hacky solution */
+    localcomm_ptr->seq = -1;
     mpi_errno = MPIR_Comm_commit(localcomm_ptr);
+    localcomm_ptr->seq = 0;
```

Comment:
So basically this localcomm gets a potentially less than ideal vci mapping? Seems OK to me.

Reply:
Yeah, I don't think anyone really care about inter-comm performance. It is not worse than global lock anyway.